File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        ArrayDeque<Integer> stack=new ArrayDeque<>();
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int num:nums2){
            while(!stack.isEmpty() && num>stack.peek()){
                int x=stack.pop();
                map.put(x,num);
            }
            stack.push(num);
          
            }
        
         while(!stack.isEmpty()){
              int y=  stack.pop();
                map.put(y,-1);
                } 
                int [] result=new int[nums1.length];
                for(int i=0;i<nums1.length;i++){
           
             result[i]=map.get(nums1[i]);
            
        }
        return result;
    }
}

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        stack=[]
        d={}
        result=[]
        for num in nums2:
            while stack and num>stack[-1]:
                smaller=stack.pop()
                d[smaller]=num
            stack.append(num)
        while stack:
            element=stack.pop()
            d[element]=-1
        for num in nums1:
            result.append(d[num])
        return result